django/db/models/fields/related.py

```diff
@@ -1561,7 +1566,9 @@ def contribute_to_class(self, cls, name, **kwargs):
         # automatically. The funky name reduces the chance of an accidental
         # clash.
         if self.remote_field.symmetrical and (
-                self.remote_field.model == "self" or self.remote_field.model == cls._meta.object_name):
+            self.remote_field.model == RECURSIVE_RELATIONSHIP_CONSTANT or
+            self.remote_field.model == cls._meta.object_name
+        ):
```

Comment:
This change is unrelated, I moved it to the #12328 and changed all hard-coded values to the constant.

Reply:
Merged in 1e0dcd6c8bfa4519c21014c73eb510620dd1a000.

Reply:
👍  rebased